core/src/main/java/io/kestra/core/models/executions/TaskRunAttempt.java

```diff
@@ -12,23 +11,27 @@
 @Value
 @Builder
 public class TaskRunAttempt {
-    @With
-    List<AbstractMetricEntry<?>> metrics;
+    /**
+     * @deprecated Should always be null, we need to keep it for backward compatibility or the deserialization of old attempt will no longer work.
+     */
+    @Deprecated
+    public void setMetrics(List<AbstractMetricEntry<?>> metrics) {
+    }
 
     @NotNull
     State state;
 
     public TaskRunAttempt withState(State.Type state) {
         return new TaskRunAttempt(
-            this.metrics,
             this.state.withState(state)
         );
     }
 
+    /**
+     * @deprecated Will always return an empty optional as metrics are now stored separatly.
+     */
+    @Deprecated
```

Comment:
You can safely remove completely

Reply:
no it's the methods `findMetrics` only used on unit test ;)

Reply:
Oh, yes, I can remove it

Reply:
Removed